src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -244,5 +244,35 @@ public static bool TryWrite<T>(Span<byte> destination, ref T value)
             Unsafe.WriteUnaligned<T>(ref GetReference(destination), value);
             return true;
         }
+
+        /// <summary>
+        /// Creates a new memory over the portion of the pre-pinned target array beginning
+        /// at 'start' index and ending at 'end' index (exclusive).
+        /// </summary>
+        /// <param name="array">The pre-pinned target array.</param>
+        /// <param name="start">The index at which to begin the memory.</param>
+        /// <param name="length">The number of items in the memory.</param>
+        /// <remarks>Returns default when <paramref name="array"/> is null.</remarks>
```

Comment:
I think we should a warning to the remarks saying that calling this method on an array that is not actually pinned is very unsafe and might result in memory corruption.